src/autoschedulers/adams2019/test.cpp

```diff
@@ -1,7 +1,58 @@
 #include "Halide.h"
+#include <cstdlib>   // setenv (or Windows _putenv_s)
+#include <iostream>  // std::cerr / std::endl
+#include <map>       // std::map
+#include <string>    // std::to_string
 
 using namespace Halide;
 
+void set_env_variable(const std::string &name, const std::string &value, int overwrite) {
+#ifdef _MSC_VER
+    _putenv_s(name.c_str(), value.c_str());
+#else
+    setenv(name.c_str(), value.c_str(), overwrite);
+#endif
+}
+
+bool test_caching(Pipeline &p1, Pipeline &p2, const Target &target, const MachineParams &params) {
+    static const std::string seed_value = Internal::get_env_variable("HL_SEED");
+    if (seed_value.empty()) {
+        // If HL_SEED is not set, then set seed for both autoscheduling executions.
+        int seed = (int)time(nullptr);
+        set_env_variable("HL_SEED", std::to_string(seed), /* overwrite */ 0);
+    }
+
+    // Turn off caching.
+    set_env_variable("HL_DISABLE_MEMOIZED_FEATURES", "1", /* overwrite */ 1);
+    set_env_variable("HL_DISABLE_MEMOIZED_BLOCKS", "1", /* overwrite */ 1);
+    auto results_without_caching = p1.auto_schedule(target, params);
+
+    // Turn on caching.
+    set_env_variable("HL_DISABLE_MEMOIZED_FEATURES", "0", /* overwrite */ 1);
+    set_env_variable("HL_DISABLE_MEMOIZED_BLOCKS", "0", /* overwrite */ 1);
+    auto results_with_caching = p2.auto_schedule(target, params);
+
+    // Reset environment variables to what they were before (memoization variables are reset in main).
+    if (seed_value.empty()) {
+        // Re-empty seed.
+        set_env_variable("HL_SEED", "", /* overwrite */ 1);
+    }
+
+    // Compare produced schedules.
```

Comment:
Very minor: this is comparing the produced features, not schedules.

Reply:
Fixed in 169fb74